float32 kp
float32 ki
float32 kd
float32 min_error_limit
float32 max_error_limit
float32 min_control_value
float32 max_control_value